### ISSUE TITLE
Bare CONTINUATION frames in section 5.1 tests

### ISSUE THREAD

**Issue by shane-kearns** [summerwind/h2spec#32]
Section 5.1 includes tests that send CONTINUATION frames on a closed or halfclosed stream.
However these CONTINUATION frames are not associated with any HEADERS or PUSH_PROMISE frame, because the preceding frame had the END_HEADERS flag set.

RFC7540 section 6.10 has this to say:
```A CONTINUATION frame MUST be preceded by a HEADERS, PUSH_PROMISE or
CONTINUATION frame without the END_HEADERS flag set.  A recipient
that observes violation of this rule MUST respond with a connection
error (Section 5.4.1) of type PROTOCOL_ERROR.```

Therefore I think a connection error with PROTOCOL error is the expected response for these test cases, rather than STREAM_CLOSED.

Also worth noting this text in section 5.1:
```Note that this diagram shows stream state transitions and the frames
and flags that affect those transitions only.  In this regard,
CONTINUATION frames do not result in state transitions; they are
effectively part of the HEADERS or PUSH_PROMISE that they follow.```

The test case "half closed (remote): Sends a CONTINUATION frame" also causes a COMPRESSION_ERROR with my implementation, because the compressed headers are truncated in the HEADERS frame with END_HEADERS + END_STREAM flags.

**Comment by tatsuhiro-t** [comment]
I submitted PR #31 to address these concerns, and it is now merged.
In #31, I just added PROTOCOL_ERROR as valid error code as well as STREAM_CLOSED.  Perhaps, we should get rid of STREAM_CLOSED?

**Comment by shane-kearns** [comment]
Your change fixes these tests - thank you.

I can see the exact error code being implementation dependent here - if the implementation checks stream ID first, it might send STREAM_CLOSED, while if it checks for the CONTINUATION frame having a HEADERS / PUSH_PROMISE to continue first, then PROTOCOL_ERROR.

**Comment by shane-kearns** [comment]
Fixed by PR #31

**Comment by summerwind** [comment]
Sorry for the late reply. #31 was landed in v1.1.1.
Thank you for the feedback!
